plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java

```diff
@@ -697,6 +699,22 @@ private String setupMountPoint(String parent) {
         return mountPoint;
     }
 
+    @Override
+    public synchronized void setSystemVmTmpltPermission(String mountPoint) {
+        if (!s_isSystemVmTmpltPermissionSet) {
+            s_logger.debug("Set permissions for " + mountPoint);
+            String result = null;
+            Script script = new Script(true, "chmod", _timeout, s_logger);
+            script.add("-R", "0777", mountPoint);
+            result = script.execute();
+            if (result != null) {
+                s_logger.warn("Unable to set permissions for " + mountPoint + " due to " + result);
+            } else {
+                s_isSystemVmTmpltPermissionSet = true;
+            }
```

Comment:
instance write-access to a static, even when in a synchronised block of code does not make for a good pattern. There is no guarantee that some other (synchronised) block accesses it as well.